src/Core/CMakeLists.txt

```
@@ -29,7 +28,7 @@ add_definitions(${OpenMP_CXX_FLAGS})
 
 
 if (WIN32)
-	set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /Ddll_EXPORTS")
+  add_definitions("/Ddll_EXPORTS")
```

presumably this is better; /CC @gfardell @paskino

---

Maybe better, however the previous line seems to be C++ specific. Having said that, I think it's ok.

---

it's used in https://github.com/TomographicImaging/CIL/blob/17e80b694303b31ac79ed25c279d3604dfa682e0/src/Core/include/dll_export.h

```suggestion
  add_compile_definitions(dll_EXPORTS)
```

---

if it's used only in `dll_export.h`, in my opinion it can be removed.
https://github.com/TomographicImaging/CIL/blob/17e80b694303b31ac79ed25c279d3604dfa682e0/src/Core/include/dll_export.h#L23-L31 has some `ifdef` stuff that should be either in CMake or the .h, but not both. The next line https://github.com/TomographicImaging/CIL/blob/17e80b694303b31ac79ed25c279d3604dfa682e0/src/Core/include/dll_export.h#L40-L42 could just as well be in merged in the `_MSC_VER` ifdef` below.

Of course, this has nothing to do with this PR.

---

all this will be purged later in #2147 <- #1875 anyway